Competition2017/src/competition/subsystems/drive/commands/DriveForDistanceCommand.java

```diff
@@ -100,10 +100,14 @@ public double calculateHeadingPower() {
         return rotationalPower;
     }
     
+    private double getYDistance() {
+        return poseSubsystem.getRobotOrientedTotalDistanceTraveled().y;
+    }
+    
     @Override
     public boolean isFinished() {
-        double velocity = driveSubsystem.getDistance() - previousPositionInches;
-        previousPositionInches = driveSubsystem.getDistance();
+        double velocity = getYDistance() - previousPositionInches;
```

Comment:
might be handy for Pose to expose velocity since it has everything internally including stuff being constantly called so could be accurate the first loop tick too?